SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -528,6 +528,11 @@ theorem one_shiftLeft_mul {x y : BitVec w} :
     1#w <<< x.toNat * y = y <<< x.toNat := by
   simp [←BitVec.mul_twoPow_eq_shiftLeft, BitVec.mul_comm]
 
+@[simp]
+theorem neg_one_mul {x : BitVec w} :
+    x.mul (-1) = (0#w).sub x := by
+  simp [BitVec.mul_comm]
+
 end BitVec
```

Comment:
```suggestion 
 
 @[simp]
theorem neg_one_mul {x : BitVec w} :
    x * BitVec.allOnes w = -x := by
  simp [← BitVec.negOne_eq_allOnes]

end BitVec
```

Reply:
To be able to put a `@[simp]` decorator, we need to make sure that the RHS is strictly simpler than the RHS. I feel this is not true for the theorem you choose, but there is another theorem after `alive_auto` has run that where a simp rewrite will lead to a clear simplification. I added this for you.

Reply:
If my comments make sense, you can commit to the suggestion and merge the PR.

Thank you for working on this. This really helps to develop a joint language.

Reply:
thanks! this also clarifies the use of `@[simp]`